lib/internal_metric/global.rb

```diff
@@ -368,10 +368,21 @@ def calc_postgres_highest_sequence
 
       RailsMultisite::ConnectionManagement.each_connection do |db|
         result[{ db: db }] = DB.query_single(<<~SQL)[0]
-          SELECT last_value
-          FROM pg_sequences
-          ORDER BY last_value DESC NULLS LAST
-          LIMIT 1
+          WITH columns_and_sequences AS (
+            SELECT table_name,
+                   column_name,
+                   information_schema.columns.data_type column_type,
+                   pg_sequences.data_type sequence_type,
```

Comment:
It doesn't seem like we're doing anything with this `sequence_type` information.

How about doing

```
WHERE column_type = 'integer' OR sequence_type = 'integer'
```
That way, we'll catch cases even where there is a mismatch.

I know we've done work to remove all known mismatches. But given that we already have the information available, maybe it would be a nice safety-net?

Reply:
I left `sequence_type` just because I think it will be useful for future devs looking into this query to have all the info in one place.

I like your suggestion and even though `column_type == sequence_type` for all the tables from core, plugins may introduce issues.